From 0350aae0dc56c32ad61dcafc665cb36a240ee6fb Mon Sep 17 00:00:00 2001
From: Puunithaaraj Gopal <puunithaaraj.gopal@intel.com>
Date: Wed, 29 May 2019 23:25:33 +0800
Subject: [PATCH 1464/1552] media: i2c: crlmodule-lite: Add ICI_FORMAT_RGB24 as
 supported format for adv7481 hdmi

ICI_FORMAT_RGB888 is currently set for RGB 32-bit and this cause issue
for FE capture as the output is RGB 24-bit. Hence a new format for
RGB 24-bit need to define to address and add it as supported format
for adv7481 hdmi input.

Tracked-On: HSD-1507216987
Tracked-On: PKT-2253
Signed-off-by: Puunithaaraj Gopal <puunithaaraj.gopal@intel.com>
---
 drivers/media/i2c/crlmodule-lite/crl_adv7481_hdmi_configuration.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/media/i2c/crlmodule-lite/crl_adv7481_hdmi_configuration.h b/drivers/media/i2c/crlmodule-lite/crl_adv7481_hdmi_configuration.h
index 489a3eb..8c2c3d4 100644
--- a/drivers/media/i2c/crlmodule-lite/crl_adv7481_hdmi_configuration.h
+++ b/drivers/media/i2c/crlmodule-lite/crl_adv7481_hdmi_configuration.h
@@ -836,6 +836,14 @@ static struct crl_csi_data_fmt adv7481_hdmi_crl_csi_data_fmt[] = {
 		.regs_items = ARRAY_SIZE(adv7481_hdmi_mode_rgb888),
 		.regs = adv7481_hdmi_mode_rgb888,
 	},
+	{
+		.code = ICI_FORMAT_RGB24,
+		.pixel_order = CRL_PIXEL_ORDER_GRBG,
+		.bits_per_pixel = 24,
+		.regs_items = ARRAY_SIZE(adv7481_hdmi_mode_rgb888),
+		.regs = adv7481_hdmi_mode_rgb888,
+	},
+
 };
 
 static struct crl_ctrl_data adv7481_hdmi_ctrls[] = {
-- 
2.7.4

